src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2650,12 +2650,24 @@ protected bool IsTaskReturningAsyncMethod()
             return symbol?.Kind == SymbolKind.Method && ((MethodSymbol)symbol).IsAsyncReturningTask(this.Compilation);
         }
 
+        protected bool IsTaskReturningAsyncMethodViaOverride()
+        {
+            var symbol = this.ContainingMemberOrLambda;
+            return symbol?.Kind == SymbolKind.Method && ((MethodSymbol)symbol).IsAsyncReturningTaskViaOverride(out _);
+        }
+
```

Comment:
Are these two methods ever called separately? If not, consider combining the methods. Same question for pair of methods below.